.github/workflows/deploy-staging.yml

```diff
@@ -12,15 +18,17 @@ env:
 
 jobs:
   deploy-tag:
-    name: Deploy to STAGING v3
+    name: Deploy to staging environment
     runs-on: [self-hosted, Linux, X64]
-    if: startsWith(github.ref, 'refs/tags/v') || startsWith(github.ref, 'refs/heads/develop') || startsWith(github.ref, 'refs/heads/v3-hotfix-release') || startsWith(github.ref, 'refs/heads/v3-release') || startsWith(github.ref, 'refs/heads/v3-hotfix/')
+    # Only run the workflow to be manually run for tags
+    # and PRs raised from release branches to main
+    if: startsWith(github.ref, 'refs/tags/v') || startsWith(github.ref, 'refs/heads/hotfix-release') || startsWith(github.ref, 'refs/heads/release')
```

Comment:
Please add develop to the list. There might be occasions when we want to deploy to staging.

Reply:
`staging` is strictly for testing the to-be-released or previously released versions. For develop branch, we already have the dev environment deployment.

What are the extra needs?

Reply:
We only have Prod and staging to test the current changes with the deployed sanity suite.

Reply:
Isn't that enough? 
In what cases do we want to deploy the `develop` branch for sanity?